application/src/main/java/org/thingsboard/server/service/edge/EdgeKafkaLagReporter.java

```diff
@@ -0,0 +1,163 @@
+import org.thingsboard.server.common.data.kv.BasicTsKvEntry;
+import org.thingsboard.server.common.data.kv.LongDataEntry;
+import org.thingsboard.server.common.data.kv.TimeseriesSaveResult;
+import org.thingsboard.server.common.data.kv.TsKvEntry;
+import org.thingsboard.server.dao.timeseries.TimeseriesService;
+import org.thingsboard.server.queue.kafka.TbKafkaSettings;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConditionalOnProperty(prefix = "queue", value = "type", havingValue = "kafka")
+public class EdgeKafkaLagReporter {
+    private static final String LAG_KEY = "downlinkLag";
+
+    private final TbKafkaSettings kafkaSettings;
+    private final TimeseriesService tsService;
+
+    @Value("${queue.kafka.lag-report.send-ts-interval-ms:60000}")
+    private long reportIntervalMs;
+
+    private KafkaConsumer<String, byte[]> consumer;
+    private ScheduledExecutorService scheduler;
+
+    @PostConstruct
+    public void init() {
+        log.info("Initializing EdgeKafkaLagReporter with interval {}ms...", reportIntervalMs);
+        this.consumer = new KafkaConsumer<>(kafkaSettings.toConsumerProps(null));
+        this.scheduler = Executors.newSingleThreadScheduledExecutor();
+        this.scheduler.scheduleAtFixedRate(this::reportLag, 0, reportIntervalMs, TimeUnit.MILLISECONDS);
+    }
+
+    private void reportLag() {
+        try {
+            AdminClient adminClient = kafkaSettings.getAdminClient();
+            Set<String> groupIds = adminClient.listConsumerGroups()
+                    .all()
+                    .get(10, TimeUnit.SECONDS)
+                    .stream()
+                    .map(ConsumerGroupListing::groupId)
+                    .filter(id -> id.startsWith("tb_edge_event.notifications."))
+                    .collect(Collectors.toSet());
+
+            log.debug("Found {} consumer groups with prefix 'tb_edge_event.notifications.'", groupIds.size());
+
+            for (String groupId : groupIds) {
+                log.debug("Processing consumer group: {}", groupId);
+
+                ListConsumerGroupOffsetsResult offsetsResult = adminClient.listConsumerGroupOffsets(groupId);
+                Map<TopicPartition, OffsetAndMetadata> groupOffsets = offsetsResult.partitionsToOffsetAndMetadata().get(5, TimeUnit.SECONDS);
+                Map<TopicPartition, Long> endOffsets = consumer.endOffsets(groupOffsets.keySet());
+
+                for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : groupOffsets.entrySet()) {
+                    TopicPartition tp = entry.getKey();
+                    String topic = tp.topic();
+                    long committed = entry.getValue().offset();
+                    Long latest = endOffsets.get(tp);
```

Comment:
let's rename latest to end or endOffset to be in sync with Kafka naming conventions